### PR TITLE
PS-5824: MTS STOP SLAVE takes too long when master crashed during event logging (8.0)

### PR DESCRIPTION
**Pull request by inikep**
https://jira.percona.com/browse/PS-5824

***
Bug #96400 : MTS STOP SLAVE takes over a minute when master crashed during event logging
https://bugs.mysql.com/bug.php?id=96400

***
FB8-105: Stop slave immediately in MTS if partial trx in the relay log can be rollbacked
https://jira.percona.com/browse/FB8-105

***
In MTS stop slave can take a minute to complete if the last transaction
is partially downloaded from the master. The slave waits for a minute for the
master to send the rest of the trx and then finally gives up. Strictly, this
wait is only required if the partial trx cannot be rollbacked safely (e.g. trx
on non-transactional engine, DDLs etc.). This change checks if there are no jobs
queued in the worker threads and if the partial trx can be rollbacked, if yes,
it immediately stops the slave.

***
This commit is based on patches from Facebook at:
- https://github.com/facebook/mysql-5.6/commit/cd39ae1
- https://github.com/facebook/mysql-5.6/commit/0623a07
with improvements from Percona at:
- https://github.com/facebook/mysql-5.6/commit/2f459f7

### CODE REVIEW

File: mysql-test/suite/rpl_gtid/t/rpl_stop_slave_partial_trx-master.opt [hunk 1]
```diff
@@ -0,0 +1 @@
+--slave_parallel_workers=8 --log-slave-updates --debug_sync_timeout=20
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl_gtid/t/rpl_stop_slave_partial_trx-master.opt]: `--log-slave-updates` is ON by default in 8.0

File: mysql-test/suite/rpl_gtid/t/rpl_stop_slave_partial_trx-slave.opt [hunk 2]
```diff
@@ -0,0 +1 @@
+--slave_parallel_workers=8 --log-slave-updates
```
Comment (percona-ysorokin) [inline mysql-test/suite/rpl_gtid/t/rpl_stop_slave_partial_trx-slave.opt]: `--log-slave-updates` is ON by default in 8.0

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Comment by inikep** [comment]
https://ps80.cd.percona.com/job/percona-server-8.0-param/313

**Pull request merged**